pkg/injectionwebhook/webhook.go

```diff
@@ -122,14 +122,11 @@ func (whsvr *WebhookServer) mutate(ar *v1beta1.AdmissionReview) (admissionRespon
 	glog.Errorf("api=mutate, message=new AdmissionReview, Kind=%v, Namespace=%v, Name=%v (%v), UID=%v, patchOperation=%v, UserInfo=%v",
 		req.Kind, req.Namespace, req.Name, pod.Name, req.UID, req.Operation, req.UserInfo)
 
-	sidecarConfig, err := sidecarconfig.RenderTemplate(corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Annotations: pod.Annotations,
-		},
-		Spec: corev1.PodSpec{
-			ServiceAccountName: pod.Spec.ServiceAccountName,
-		},
-	}, whsvr.sidecarConfigTemplate)
+	// the pod might not have its namespace defined since it's still in admission,
+	// let's insert it from the req to expose as much data to templating as possible
+	pod.Namespace = req.Namespace
```

Comment:
Great, as I said, there some projects using this approach, take a look https://github.com/search?q=%22pod.Namespace+%3D+req.Namespace%22&type=code